src/Swarm/Util.hs

```diff
@@ -259,7 +259,7 @@ getDataFileNameSafe name = do
 
 dataNotFound :: FilePath -> IO Text
 dataNotFound f = do
-  d <- getSwarmDataPath False
+  d <- getSwarmDataPath False ""
```

Comment:
I take it that the empty string represents "no subdirectory", i.e. the "current directory"?  Perhaps a `Maybe` would convey the intentions better.

Reply:
It's how path works - `a//./` is the same as `a`.

A good solution might be to use a different function or make a proper ADT type for our sources. 🤔 

We should refactor this code and document it better, but let’s do that in a new PR.